frontend/src/bundles/common/pages/budgets/budget-details/components/budget-modal/budget-modal.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useCallback } from 'react';
+import { Controller } from 'react-hook-form';
+
+import { type BudgetCreateRequestDto } from '~/bundles/budgets/budgets';
+import { actions as budgetsActions } from '~/bundles/budgets/store';
+import {
+    BaseModal,
+    Button,
+    Input,
+} from '~/bundles/common/components/components';
+import { ButtonSize } from '~/bundles/common/enums/button-size.enum';
+import { ButtonVariant } from '~/bundles/common/enums/button-variant.enum';
+import { useAppDispatch, useAppForm } from '~/bundles/common/hooks/hooks';
+
+import {
+    RenderCurrency,
+    RenderDate,
+    RenderMultiDropdown,
+    RenderRecurrence,
+} from '../components';
+import styles from './styles.module.scss';
+
+interface BudgetModalProperties {
```

Comment:
We use types in components props typization and i recomend to use just just "Properties".

Reply:
this is the code from another branch